client/landing/gutenboarding/onboarding-block/style-preview/font-select.tsx

```diff
@@ -14,19 +14,20 @@ import { useI18n } from '@automattic/react-i18n';
  */
 import { fontPairings, getFontTitle, FontPair } from '../../constants';
 import { STORE_KEY } from '../../stores/onboard';
-import designs from '../../available-designs';
+import type { Design } from '../../stores/onboard/types';
 
 const FontSelect: React.FunctionComponent = () => {
 	const { __ } = useI18n();
 	const { selectedDesign, selectedFonts } = useSelect( ( select ) =>
 		select( STORE_KEY ).getState()
 	);
+	const { getRandomizedDesigns } = useSelect( ( select ) => select( STORE_KEY ) );
```

Comment:
Any idea why line 21 uses a different pattern, with the `.getState()` call at the end?

I noticed inconsistency with the store name, here `STORE_KEY` is used but in client/landing/gutenboarding/onboarding-block/design-selector/index.tsx `ONBOARD_STORE` is used.    

And then another difference,  the persistent store is called `WP_ONBOARD`, though I could understand a motivation for calling the persistent store a different name.

Reply:
I think it might be that historically in Gutenboarding we only had a single store, so referred to it as `STORE_KEY`, then when additional stores were added (e.g. `PLANS_STORE`) we needed to distinguish between them. In this file it would make sense to use the same `import { STORE_KEY as ONBOARD_STORE };` line as we're using elsewhere in Gutenboarding. You're welcome to change it in this PR if you'd like (or leave it as is, whichever you prefer).

I _think_ the `WP_` prefix for the persistence is probably going with the pattern borrowed from Gutenberg of using that prefix: https://github.com/wordpress/gutenberg/blob/HEAD/packages/data/src/plugins/persistence/index.js#L38 but I could be wrong!

Reply:
Coolio, I'll make the change to use ` import { STORE_KEY as ONBOARD_STORE };` for consistency.

Reply:
@andrewserong This change has a bit of a ripple effect with respect to typings.  
In this font-select.tsx file, I would be replacing 
```
	const { selectedDesign, selectedFonts } = useSelect( ( select ) =>
		select( STORE_KEY ).getState()
	);
```
with 
```
	const { getRandomizedDesigns, getSelectedDesign, getSelectedFonts } = useSelect( ( select ) =>
		select( ONBOARD_STORE )
	); 
```
and thus replacing all instances of `selectedDesign` with `getSelectedDesign()` and  replacing `selectedFonts` with `getSelectedFonts()`.  

If we look at `selectedFonts` in this line https://github.com/Automattic/wp-calypso/blob/f3a765b2b5fb0707b22f85abba0b489ccede78ce/client/landing/gutenboarding/onboarding-block/style-preview/font-select.tsx#L50
when I update that with the selector function `getSelectedFonts()` , typescript complains (see screenshots below) about the potential `undefined` values.
What I don't understand is that the selector is just returning the reducer value! The `selectedFonts` [reducer](https://github.com/Automattic/wp-calypso/blob/f3a765b2b5fb0707b22f85abba0b489ccede78ce/client/landing/gutenboarding/stores/onboard/reducer.ts#L126) is typed to return `FontPair | undefined` (assuming I'm understanding all this correctly).   Do you have an idea about why Typescript was not complaining about a possible undefined state before I made the change to use `getSelectedFonts`?  


![Screen Shot 2020-08-21 at 12 22 16](https://user-images.githubusercontent.com/5665959/90845641-c73f1600-e3a9-11ea-8c2a-75526d254cdd.png)

![Screen Shot 2020-08-21 at 12 21 42](https://user-images.githubusercontent.com/5665959/90845652-cc9c6080-e3a9-11ea-8a38-0dcc00da2981.png)

Reply:
Good question! For the TypeScript problem, it looks like that line is currently covered by a check that `selectedFonts` is not undefined (e.g. https://github.com/automattic/wp-calypso/blob/83dbb0fa46689e25608974d5f0768b47a25d8fd3/client/landing/gutenboarding/onboarding-block/style-preview/font-select.tsx#L46), so when the `headings` key is accessed, that code path can only run if it's really available. However if you replace that check with `getSelectedFonts()`, TypeScript can't know that the next time you call it, the return of that function is definitely going to be the same. From TypeScript's perspective, that function could still potentially return `undefined`. In principle this is a good thing, because if the behaviour of the `getSelectedFonts` function/selector were to change in the future, we'd need to know this part of the code is still safe.

To keep your changeset small, you could update the name of the constant for the store, but otherwise leave the useSelect call as is, and add your line directly after it. E.g.

```
const { selectedDesign, selectedFonts } = useSelect( ( select ) =>
		select( ONBOARD_STORE ).getState()
	);
const { getRandomizedDesigns } = useSelect( ( select ) => select( ONBOARD_STORE ) );
```

However it also might make sense to group these together. If you do, on the line you're currently looking at, you wouldn't want to call `getFontTitle` with an `undefined` value, so I think this part of the component might still need to be guarded by the `selectedFonts` check. I guess you could add a line before `selectedFontOption` — `const selectedFonts = getSelectedFonts();` —  but either way there's a bit of duplication!

These sorts of little optimisations can sometimes be tricky! My preference is usually to try to change the smallest amount possible that's needed for the change I'm proposing in a PR, and then refactor as necessary in a follow-up janitorial PR. However, sometimes it feels messier to do it that way, so if it makes more sense to change these calls, then by all means, cleaning as you go is good, too!

Reply:
Thanks for that clarification with regard to `const selectedFontOption = selectedFonts ? ( ....`:
>  it looks like that line is currently covered by a check that selectedFonts is not undefined 

Now the typescript issue makes sense.  💡 

Now or next :), with a little more awareness under my belt, I did a search for `select( ONBOARD_STORE )` and `.getState()` and I can see there is a mix of usage of calling `.getState()` and using reducers vs not calling `.getState()`  and using selectors, so I _will_ update the `ONBOARD_STORE` name but I _will not_' make the other changes that replace the reducer with the selectors since both patterns are used throughout gutenboarding.

![Screen Shot 2020-08-21 at 15 44 15](https://user-images.githubusercontent.com/5665959/90856701-e5b30a80-e3c5-11ea-8325-146fc94cc52d.png)

![Screen Shot 2020-08-21 at 15 44 55](https://user-images.githubusercontent.com/5665959/90856727-f2cff980-e3c5-11ea-8fcd-9ecd31ce0862.png)

That was all great exploration to understand reducers and selectors!!! 💡 💡